example/lib/main.dart

```diff
@@ -122,163 +157,226 @@ class _HomePageState extends State<HomePage> {
         child: Column(
           mainAxisAlignment: MainAxisAlignment.spaceEvenly,
           children: [
-            Board(
-              size: screenWidth,
-              settings: BoardSettings(
-                pieceAssets: pieceSet.assets,
-                colorScheme: boardTheme.colors,
-                enableCoordinates: true,
-                animationDuration: pieceAnimation
-                    ? const Duration(milliseconds: 200)
-                    : Duration.zero,
-                dragFeedbackSize: dragMagnify ? 2.0 : 1.0,
-                drawShape: DrawShapeOptions(
-                  enable: drawMode,
-                  onCompleteShape: _onCompleteShape,
-                  onClearShapes: () {
-                    setState(() {
-                      shapes = ISet();
-                    });
-                  },
+            if (playMode == Mode.boardEditor)
+              Column(children: [
+                makePieceEditorMenu(Side.black, screenWidth),
+                const SizedBox(height: 20),
+                BoardEditor(
+                  size: screenWidth,
+                  orientation: orientation,
+                  fen: writeFen(boardEditorPieces),
+                  settings: BoardEditorSettings(
+                    pieceAssets: pieceSet.assets,
+                    colorScheme: boardTheme.colors,
+                    enableCoordinates: true,
+                  ),
+                  onTappedSquare: (squareId) => setState(() {
+                    boardEditorPieces[squareId] = Piece(
```

Comment:
would need to implement the "bin" button to remove pieces by tap later.

Reply:
I'm not even sure if we actually want the bin button for the mobile v2 app? I took a look at the old mobile app, which does not have it either (pieces are discarded by dragging them off the board, also there' not tap-to-add-piece feature). Do you think we still want tap-to-remove and tap-to-add in mobile v2? Or is drag-and-drop enough?

Reply:
Yeah I think we want both. Same as the lichess web UI. It is easy to implement so it's not a big deal imo.

And it would be nice to have it in the chessground demo app too, because it shows the capabilities of the board editor.